File: VB/ExcelDataBaseHelper.vb
Imports System.IO
Imports System.Data.OleDb
Imports System.Data
Imports DevExpress.XtraEditors

Namespace WindowsApplication1

    Public Class ExcelDataBaseHelper

        Public Shared Function OpenFile(ByVal fileName As String) As Object
            Dim fullFileName = String.Format("{0}\{1}", Directory.GetCurrentDirectory(), fileName)
            If Not File.Exists(fullFileName) Then
                XtraMessageBox.Show("File not found")
                Return Nothing
            End If

            Dim connectionString = String.Format("Provider=Microsoft.Jet.OLEDB.4.0; data source={0}; Extended Properties=Excel 8.0;", fullFileName)
            Using adapter = New OleDbDataAdapter("select * from [Sheet1$]", connectionString)
                Dim ds = New DataSet()
                Dim tableName As String = "excelData"
                adapter.Fill(ds, tableName)
                Dim data As DataTable = ds.Tables(tableName)
                Return data
            End Using
        End Function
    End Class
End Namespace

File: CS/ExcelDataBaseHelper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;
using System.Data.OleDb;
using System.Data;

namespace WindowsApplication1
{
    public class ExcelDataBaseHelper
    {
        public static object OpenFile(string fileName)
        {
            var fullFileName = string.Format("{0}\\{1}", Directory.GetCurrentDirectory(), fileName);
            if (!File.Exists(fullFileName))
            {
                System.Windows.Forms.MessageBox.Show("File not found");
                return null;
            }
            var connectionString = string.Format("Provider=Microsoft.Jet.OLEDB.4.0; data source={0}; Extended Properties=Excel 8.0;", fullFileName);
            var adapter =  new OleDbDataAdapter("select * from [Sheet1$]", connectionString);
            var ds = new DataSet();
            string tableName = "excelData";
            adapter.Fill(ds, tableName);
            DataTable data = ds.Tables[tableName];
            return data;
        }
        
    }
}
